### ISSUE TITLE
add default export to allow for usage from ECMAScript 6

### ISSUE THREAD

**Issue by michaelficarra** [jashkenas/underscore#2253]
This is needed to allow for the `_` in the following program:

```js
import _, {some, every} from "underscore";
```

PR incoming.

**Comment by akre54** [comment]
You mean adding a `default` property and `__esModule` onto the exports object? Babel and others already [wrap imports](http://babeljs.io/repl/#?experimental=true&evaluate=true&loose=true&spec=false&playground=false&code=import%20_%2C%20%7Bsome%2C%20every%7D%20from%20%22underscore%22%3B) for compatibility.

**Comment by michaelficarra** [comment]
But that's a babel/traceur/etc. quirk, and `__esModule` is not spec compatible. In order to use the `commonStrict` module format in Babel and be [fully conformant with ES6 modules](https://github.com/babel/babel/issues/675), underscore should export `default`.

**Comment by jashkenas** [comment]
On what platforms do "ES6" modules exist at this point? I can't find them in Kangax's tables.

**Comment by michaelficarra** [comment]
@jashkenas None. Most are blocked on defining a loader, something TC39 punted to the runtimes. But Babel has defined multiple loaders that can be chosen via the `--modules` flag, and `commonStrict` is the option that is future compatible. See the [section on interop](https://babeljs.io/docs/usage/modules/#interop). So anyone using this module loader *or* anyone using an ES6 module implementation with CommonJS interop in the future will need this export.

**Comment by jdalton** [comment]
This this appears to just be a transpiler aid and not an ES6 requirement it looks out of scope for Underscore.
